### ISSUE TITLE
Do not require typehints for positional or keyword-only ( '*' or '/' ) parameters

### ISSUE THREAD

**Issue by Josverl** [sourcery-ai/sourcery#314]
### Checklist

- [x] I have searched the [Sourcery documentation](https://docs.sourcery.ai/) for the issue, and found nothing
- [x] I have checked there are no open [bugs](https://github.com/sourcery-ai/sourcery/issues?q=label%3Abug) referencing the same bug or problem

### Description

sourcery should not require typehints for positional or keyword-only ( '*' or '/' ) parameters

`Annotate parameter * in public function/method publish with a type`

![image](https://user-images.githubusercontent.com/981654/212704085-283ad0e2-82eb-4b22-8d49-ba55b4e0a6ed.png)

Note that there are a few other params that could use a type annotation, however '*' should **not** be annotated.

config:
``` yaml
rule_settings:
  enable: 
  - default
  - require-parameter-annotation
```
### Code snippet that reproduces issue

```python
def publish(
    *,
    db: PysonDB,
    pkg_type,
    version: str,
    family="micropython",
    production,  # PyPI or Test-PyPi
    dryrun=False,  # don't publish , dont save to the database
    force=False,  # publish even if no changes
    clean: bool = False,  # clean up afterwards
    port: str = "",
    board: str = "",
) -> Dict[str, Any]:
  ...
```

### Debug Information
IDE Version: 
VScode

Sourcery Version: 
Sourcery 1.0.3

Operating system and Version:
Windows 11

**Comment by gsnidero** [comment]
I found the same issue and am using: 
* Sourcery v1.0.3
* VSCode v1.74.3
* macOS 13.1 (22C65) on Apple M1

**Comment by Josverl** [comment]
Thanks for the quick fix,

Do I need to change my config in order to use this, if so; How ?

**Comment by ruancomelli** [comment]
Hey @Josverl!

It depends on how you are using the GPSG rules:

1. if you have copied the rules from the [`google-python-style-guide.yaml`](https://github.com/sourcery-ai/sourcery-rules/blob/main/google-python-style-guide.yaml) config file into a local `.sourcery.yaml` file, then you will need to edit it to include the fix I submitted to that repository: adding the extra `not arg.equals("*") and not arg.equals("/")` to the `condition` field of `require-parameter-annotation`;
2. if you are using the GPSG rules that ship with the Sourcery CLI, then the fix will become automatically available to you after that fix is merged and a new Sourcery version is released;

I will let you know once the PR is merged and also once we release the fix.

**Comment by ruancomelli** [comment]
Hello @Josverl and @gsnidero!

The fix for this issue was just merged into the https://github.com/sourcery-ai/sourcery-rules repository. The next Sourcery release (probably v1.0.5) will ship with the fix.
